### PR TITLE
Fix `$inc` operator panics for non-existing array index

### LINKED ISSUE #1658: `$inc` operator panics for non-existing path
### Versions

0.7.1

### What did you do?

Moved `TestUpdateFieldInc` test `ArrayFieldValueNotExist` test case to compat test.

```
		"ArrayFieldValueNotExist": {
			update: bson.D{{"$inc", bson.D{{"v.0.foo", int32(1)}}}},
			skip:   "https://github.com/FerretDB/FerretDB/issues/1658",
		},
```
### What did you expect to see?

Same as compat.

### What did you see instead?

Received unexpected internal error was returned.
```
panic: types.getByPath: types.Array.Get: index 0 is out of bounds [0-0)
        panic: err must be non-nil [recovered]
        panic: err must be non-nil
(err = <nil>)

```

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #1658.

Solved by https://github.com/FerretDB/FerretDB/pull/1795 so just unskip test

## Readiness checklist

* [ ] I added unit tests for new functionality or bug fixes.
* [ ] I added integration tests for new functionality or bug fixes.
* [x] I added compatibility tests for new functionality or bug fixes.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/types/path.go [hunk 1]
```diff
@@ -211,6 +211,12 @@ func insertByPath(doc *Document, path Path) error {
 			case *Document:
 				v.Set(insertedPath.Slice()[suffix], must.NotFail(NewDocument()))
 			case *Array:
+				if insertedPath.Slice()[suffix] == "0" {
+					if err := v.Append(must.NotFail(NewDocument())); err == nil {
```
Comment (chilagrow) [inline internal/types/path.go:215]: This fix is a bit questionable to me. It just does the name thing as document.

Comment (rumyantseva) [reply internal/types/path.go:215]: What if the test case is `bson.D{{"$inc", bson.D{{"v.1.foo", int32(1)}}}},`? How should we behave?

**Review by rumyantseva — COMMENTED** [review]
Asked a question (is `0` such a particular case, should we check other indexes somehow?..)

File: internal/types/path.go (hunk 1 above)
Comment (rumyantseva) [reply internal/types/path.go:215]: Ok, I had a task with a similar problem :)

I suggest this implementation: https://github.com/FerretDB/FerretDB/pull/1795/files#diff-4dba376d417720306aee8c5e84635828f8d8d9cb9dd4890d67a7827f5138c017R268-R275

File: integration/update_field_compat_test.go [hunk 2]
```diff
@@ -206,7 +206,9 @@ func TestUpdateFieldCompatIncComplex(t *testing.T) {
 		},
 		"ArrayFieldValueNotExist": {
 			update: bson.D{{"$inc", bson.D{{"v.0.foo", int32(1)}}}},
-			skip:   "https://github.com/FerretDB/FerretDB/issues/1658",
+		},
+		"ArrayFieldIndexNotExist": {
+			update: bson.D{{"$inc", bson.D{{"v.5.foo", int32(1)}}}},
```
Comment (chilagrow) [inline integration/update_field_compat_test.go:211]: Solved by https://github.com/FerretDB/FerretDB/pull/1795

**Review by rumyantseva — APPROVED** [review]
LGTM

**Review by w84thesun — APPROVED** [review]

**Pull request merged**